lib/api/pokedex_api_spec/handler/pokemon_handler.dart

```diff
@@ -1,21 +1,25 @@
 import 'package:pokedex/api/api_client.dart';
 import 'package:pokedex/api/pokedex_api_spec/model/pokemon_model.dart';
+import 'package:pokedex/utils/strings.dart' as str;
 
 class PokemonApi {
   final ApiClient apiClient;
 
   PokemonApi(this.apiClient);
 
-  static const int _limitAmount = 20;
-
-  Future<Pokemon> getPokemonList() async {
+  Future<List<Pokemon>> getPokemonList(String? offset, String? limit) async {
     final queryParams = <String, dynamic>{};
+    if (limit != null) {
+      queryParams['limit'] = limit;
+    }
+    if (offset != null) {
+      queryParams['offset'] = offset;
+    }
 
     final baseUri = Uri.parse(apiClient.dio.options.baseUrl);
-    final uri = baseUri.replace(queryParameters: queryParams, path: '${baseUri.path}/pokemon/?limit=$_limitAmount');
-
-    return apiClient.dio
+    final uri = baseUri.replace(queryParameters: queryParams, path: '${baseUri.path}/${str.pokemonRoute}/');
+    return await apiClient.dio
         .getUri(uri)
-        .then((response) => response.data.map((dynamic json) => Pokemon(name: json['name'])));
+        .then((response) => response.data[str.results].map<Pokemon>((pokemon) => Pokemon.fromJson(pokemon)).toList());
```

Comment:
Explain to me what .map did here

Reply:
it mapped the json string from the results and cast it into a pokemon object

Reply:
Explain to me in detail 😂 I want to know if you really understand what it does xD

Reply:
I got the Json string which is response.data and I mapped the field results which has name and url and cast it to the model Pokemon and called the fromjson which assigns the name and url and make them into a list thru toList() function and loop through the pokemons

Reply:
I transformed the json string that I got into my desirable type which is List of Pokemon by using the jsonserialization and the toList() function and iterate thru the list and at the same time creating a Pokemon object

Reply:
Also, I used map as it is a built in function that is much more efficient to use because you wont need to add more lines and add one object at a time and the lesser the lines the more readable it will be.

Reply:
- What is .map?
.map is a built-in function for flutter that lets you iterate through the list and do something to the data and later on transform the data into the desired type.
- What is the purpose of .map?
it gives you the freedom to change the data at each iteration and also transform the data into the desired type or outcome
- Why do you use .map over normal loop
.map is more efficient, readable, reusable compare to using normal loop because it will take lesser lines making it more readable. As to my example having to create an object and manually add it to the list would be more inefficient as to using .map and let the function do the work, there are also built in functions associated with map that would be beneficial like filter.